From b775b353a8aafedc3d06c9fb2e47d83c098caefd Mon Sep 17 00:00:00 2001
From: root <root@frontier.(none)>
Date: Mon, 20 Feb 2012 19:09:49 +0100
Subject: [PATCH] fix for kernel 3.3

---
 kernel/conftest.sh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/kernel/conftest.sh b/kernel/conftest.sh
index c7cd056..e0cda73 100755
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -126,7 +126,7 @@ build_cflags() {
     CFLAGS="$BASE_CFLAGS $MACH_CFLAGS $OUTPUT_CFLAGS -I$HEADERS $AUTOCONF_CFLAGS"
 
     if [ "$ARCH" = "i386" -o "$ARCH" = "x86_64" -o "$ARCH" = "x86" ]; then
-        CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include"
+        CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include -I$SOURCES/arch/x86/include/generated"
     elif [ "$ARCH" = "ARMv7" ]; then
         CFLAGS="$CFLAGS -I$SOURCES/arch/arm/include"
     fi
-- 
1.7.8.4

