diff --git a/Src/hist.c b/Src/hist.c
index bd03c4f..da29149 100644
--- a/Src/hist.c
+++ b/Src/hist.c
@@ -2952,7 +2952,7 @@ lockhistfile(char *fn, int keep_trying)
 	char *lockfile;
 #ifdef HAVE_LINK
 # ifdef HAVE_SYMLINK
-	char pidbuf[32], *lnk;
+	char pidbuf[64], *lnk;
 # else
 	char *tmpfile;
 # endif
@@ -2968,7 +2968,7 @@ lockhistfile(char *fn, int keep_trying)
 	 * avoid a change from open()-based locking to symlink()-based. */
 #ifdef HAVE_LINK
 # ifdef HAVE_SYMLINK
-	sprintf(pidbuf, "/pid-%ld/host-", (long)mypid);
+	sprintf(pidbuf, "/tmp/zsh-dummy-symlink-lock/pid-%ld/host-", (long)mypid);
 	lnk = getsparam("HOST");
 	lnk = bicat(pidbuf, lnk ? lnk : "");
 	/* We'll abuse fd as our success flag. */
