diff --git a/src/torrent/tracker.cc b/src/torrent/tracker.cc
index f370432..41f026e 100644
--- a/src/torrent/tracker.cc
+++ b/src/torrent/tracker.cc
@@ -49,7 +49,7 @@ Tracker::Tracker(TrackerList* parent, const std::string& url, int flags) :
   m_group(0),
   m_url(url),
 
-  m_normal_interval(1800),
+  m_normal_interval(600),
   m_min_interval(600),
 
   m_latest_event(EVENT_NONE),
diff --git a/src/torrent/tracker.h b/src/torrent/tracker.h
index 98e2bb0..dbfbc5a 100644
--- a/src/torrent/tracker.h
+++ b/src/torrent/tracker.h
@@ -140,7 +140,7 @@ protected:
 
   void                set_group(uint32_t v)                 { m_group = v; }
 
-  void                set_normal_interval(int v)            { if (v >= 600 && v <= 3600) m_normal_interval = v; }
+  void                set_normal_interval(int v)            { if (v >= 60 && v <= 600) m_normal_interval = v; }
   void                set_min_interval(int v)               { if (v >= 300 && v <= 1800) m_min_interval = v; }
 
   int                 m_flags;
