From 67e3523a2c6d2a7c8564f194878fe65c6bf8a217 Mon Sep 17 00:00:00 2001
From: Piotr Karbowski <piotr.karbowski@gmail.com>
Date: Sat, 30 Apr 2011 21:39:20 +0200
Subject: [PATCH] Does not reset in-memory db and graphs on sigusr1.

---
 darkstat.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/darkstat.c b/darkstat.c
index 6b393c7..ddf3a90 100644
--- a/darkstat.c
+++ b/darkstat.c
@@ -423,8 +423,6 @@ main(int argc, char **argv)
 
       if (reset_pending) {
          if (export_fn != NULL) db_export(export_fn); /* FIXME: USR2? */
-         hosts_db_reset();
-         graph_reset();
          reset_pending = 0;
       }
 
-- 
1.7.5.rc1

