diff --git a/pym/portage/dbapi/vartree.py b/pym/portage/dbapi/vartree.py
index 11384bf..b431669 100644
--- a/pym/portage/dbapi/vartree.py
+++ b/pym/portage/dbapi/vartree.py
@@ -4785,7 +4785,7 @@ class dblink(object):
 						encoding=_encodings['merge'])
 					if mymtime is None:
 						return 1
-					hardlink_candidates.append(mydest)
+					#hardlink_candidates.append(mydest)
 					zing = ">>>"
 
 					try:
