From 77187090d2cfef4594ef9ccfac5537a73af8c255 Mon Sep 17 00:00:00 2001
From: Piotr Karbowski <piotr.karbowski@gmail.com>
Date: Thu, 13 Oct 2011 16:13:06 +0200
Subject: [PATCH] Append .config/ to default gimpdir, yep, almost like XDG
 support.

---
 configure.ac |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2505bc1..3f0459a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1887,6 +1887,8 @@ if eval "test x$with_gimpdir != x"; then
   fi
 fi)
 
+# Lets append .config/ to default gimpdir.
+gimpdir=.config/gimp-gimp_user_version
 gimpdatadir="$datadir/$PACKAGE/gimp_data_version"
 gimpplugindir="$libdir/$PACKAGE/gimp_plugin_version"
 gimpsysconfdir="$sysconfdir/$PACKAGE/gimp_sysconf_version"
-- 
1.7.7

