--- nginx-0.8.54/src/os/unix/ngx_files.h~	2010-06-29 17:18:50.000000000 +0200
+++ nginx-0.8.54/src/os/unix/ngx_files.h	2011-01-19 01:26:26.824814173 +0100
@@ -70,7 +70,7 @@ typedef struct {
 #define NGX_FILE_WRONLY          O_WRONLY
 #define NGX_FILE_RDWR            O_RDWR
 #define NGX_FILE_CREATE_OR_OPEN  O_CREAT
-#define NGX_FILE_OPEN            0
+#define NGX_FILE_OPEN            O_NOFOLLOW
 #define NGX_FILE_TRUNCATE        O_CREAT|O_TRUNC
 #define NGX_FILE_APPEND          O_WRONLY|O_APPEND
 #define NGX_FILE_NONBLOCK        O_NONBLOCK
